core/service/src/engine/threshold/service/public_decryptor.rs

```diff
@@ -511,76 +528,75 @@ impl<
                     | FheTypes::Uint16
                     | FheTypes::Uint32
-                    | FheTypes::Uint64 => Self::inner_decrypt::<u64>(
-                        internal_sid,
-                        context_id,
-                        epoch_id,
-                        session_maker,
-                        ciphertext,
-                        fhe_type,
-                        ct_format,
-                        fhe_keys_rlock,
-                        dec_mode,
-                    )
-                    .await
-                    .map(|x| TypedPlaintext::new(x as u128, fhe_type)),
-                    unsupported_fhe_type => {
-                        anyhow::bail!("Unsupported fhe type {:?}", unsupported_fhe_type);
+                    | FheTypes::Uint64 => {
+                        RealPublicDecryptor::<PubS, PrivS, Dec>::inner_decrypt::<u64>(
+                            internal_sid,
+                            context_id,
+                            epoch_id,
+                            session_maker,
+                            ciphertext,
+                            fhe_type,
+                            ct_format,
+                            fhe_keys_rlock,
+                            dec_mode,
+                        )
+                        .await
+                        .map(|x| TypedPlaintext::new(x as u128, fhe_type))
                     }
+                    // TODO handle error here
+                    unsupported_fhe_type => Err(anyhow::anyhow!(
+                        "Unsupported fhe type {:?}",
+                        unsupported_fhe_type
+                    )),
                 };
                 // We don't update the error counter here but rather in the signature task
                 // so we only update it once even if there are multiple decryption task that fail
                 match res_plaintext {
                     Ok(plaintext) => Ok((ctr, plaintext)),
-                    Result::Err(e) => Err(anyhow_error_and_log(format!(
-                        "Threshold decryption failed:{e}"
-                    ))),
+                    Result::Err(e) => Err(anyhow::anyhow!("Threshold decryption failed: {e}")),
                 }
             };
             dec_tasks.push(
                 self.tracker
                     .spawn(decrypt_future().instrument(tracing::Span::current())),
             );
         }
-
-        // collect decryption results in async mgmt task so we can return from this call without waiting for the decryption(s) to finish
-        let meta_store = Arc::clone(&self.pub_dec_meta_store);
-        let sigkey = self.base_kms.sig_key().map_err(|e| {
-            tonic::Status::internal(format!(
-                "Failed to get signing key for public decryption request {req_id}: {e:?}"
-            ))
-        })?;
+        // TODO the code below could be simplified a lot of we don't want to log individual lock time and do so many tiny threads
```

Comment:
WDYTH @reviewer?

Reply:
Logs pruned and embedded micro benches removed in 54d00495b0480d4b552a3b88ccec3cb4b84d1ba0